Musicca/app/src/main/java/com/example/musicca/activities/CurrentPlaylistActivity.java

```diff
@@ -24,6 +24,8 @@
 import com.parse.ParseQuery;
 
 import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
```

Comment:
What is this for?

Reply:
I was trying to set up a timer that refreshes every 10 seconds but there was a warning every 10 seconds that the app was doing too much work on its main thread. I found an alternative way of refreshing the songs to ensure they remain sorted based on the number of likes. I refresh every time the user likes/ unlikes the song

Reply:
Like the warning said, you need to put the refresh code in another thread. You could still do that if you want to implement that.

Reply:
On this timer function:
// refreshes songs every 10 seconds
new Timer().schedule(new TimerTask() {
    @Override
    public void run() {
        runOnUiThread(new Runnable() {
            @Override
            public void run() {
                try {
                    currentPlaylistAdapter.updateSongs();
                } catch (ParseException ex) {
                    ex.printStackTrace();
                    Log.d(TAG, "Could not refresh songs!");
                }
            }
        });
    }
}, 10000, 1000); 
Loading views can only run on the main UI thread, hence the function runOnUiThread. I can look into views on another thread. I am currently using the approach of refreshing the list upon like/ unlike.

Reply:
I see. You don't need to implement that on this PR. If you are interested, feel free to look into that later.